test/spec/modules/BTBidAdapte_spec.js

```diff
@@ -0,0 +1,432 @@
+            siteId: '1234567',
+            pubmatic: {
+              publisherId: '111111',
+            },
+          },
+          schain: {
+            ver: '1.0',
+            complete: 1,
+            nodes: [{ asi: 'test.com', sid: '1234567', hp: 1 }],
+          },
+          transactionId: '111111',
+          userIdAsEids: [{ source: 'test.com', uids: [{ id: '11111' }] }],
+        },
+      ];
+
+      const expectedData = JSON.stringify({
+        id: 'test12345',
+        source: {
+          tid: '111111',
+          ext: {
+            schain: {
+              ver: '1.0',
+              complete: 1,
+              nodes: [{ asi: 'test.com', sid: '1234567', hp: 1 }],
+            },
+          },
+        },
+        imp: [
+          {
+            id: 'bid12345',
+            tagid: 'leaderboard',
+            ab: true,
+            siteId: '1234567',
+            ext: { pubmatic: { publisherId: '111111' } },
+            banner: { format: [{ w: 728, h: 90 }] },
+            bidfloor: 0.03,
+            bidfloorcur: 'USD',
+          },
+        ],
+        device: {
+          w: 785,
+          h: 600,
+          ua: 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) HeadlessChrome/117.0.5938.92 Safari/537.36',
+          devicetype: 'pc',
+        },
+        site: { page: 'https://test.com/', domain: 'test.com' },
+        user: {
+          ext: {
+            consent: 'gdprString',
+            eids: [{ source: 'test.com', uids: [{ id: '11111' }] }],
+          },
+        },
+        regs: {
+          ext: { gdpr: 1, us_privacy: 'uspString' },
+          gpp: 'gppString',
+          gpp_sid: [7],
+        },
+        at: 1,
+        ext: { ssl: 1 },
+        test: 1,
+      });
+
+      sandbox.stub(config, 'getConfig').callsFake((key) => {
+        const config = {
+          debug: true,
+          pageUrl: 'https://test.com/',
+        };
+        return deepAccess(config, key);
+      });
+
+      const requests = spec.buildRequests(validBidRequests, bidderRequest);
+
+      expect(requests).to.be.an('array');
+      expect(requests).to.have.lengthOf(validBidRequests.length);
+      expect(requests[0].method).to.equal('POST');
+      expect(requests[0].url).to.equal(ENDPOINT_URL);
+      expect(requests[0].data).to.equal(expectedData);
+      expect(requests[0].bids).to.equal(validBidRequests);
+    });
+  });
+
+  describe('interpretResponse', () => {
+    it('should return empty array if serverResponse is not defined', () => {
+      const bids = spec.interpretResponse(undefined, {});
+
+      expect(bids.length).to.equal(0);
+    });
+
+    it('should return empty array if seatbid array is empty', () => {
+      const serverResponse = {
+        body: {
+          seatbid: [],
+        },
+      };
+
+      const bids = spec.interpretResponse(serverResponse, {});
+
+      expect(bids.length).to.equal(0);
+    });
+
+    it('should return bids array', () => {
```

Comment:
Would it better to have this say something like: "should return bids array when serverResponse is defined and seatbid array is not empty"? Feel free to leave it as-is, I'm not really sure what the convention should be.